### ISSUE TITLE
Cross browser displsy issue (Chromium vs FireFox) sort indicator icon blocked by gridmenu button

### ISSUE THREAD

**Issue by mcgolkev** [ghiscoding/Angular-Slickgrid#1031]
### Describe the bug

In Firefox browser the slick-sort-indicator-asc icon is being blocked by the slick-gridmenu-button in the top right most grid column.  These two icons are layered on top of each other and they both work as a user can still see a piece of the sort icon to be able to click on it.  This occurs regardless of the amount of space on the screen, even with overly wide columns the menu icon blocks the sort icon.  When this application is run on Chromium, the gridmenu button appears to the right of the sort icon and both are correctly visible.

This seems to be a cross browser compatibility issue.  Are there some grid options that might cure this issue?

### Reproduction

I am not sure what to put here (new dev).

 The issue appears for any slickgrid implementation I have done, the sort icon is blocked in Firefox, it is correctly displayed in Chromium.

### Environment Info

```shell
Firefox 106.0.2
Chromium 106.0.5249.119
typescript 3.5.3
Angular 8.2.14
Op sys: linux x64
Angular Slickgrid 3
```

### Validations

- [X] Follow our [Code of Conduct](https://github.com/ghiscoding/Angular-Slickgrid/blob/master/.github/CODE_OF_CONDUCT.md)
- [X] Read the [HOWTO - Step by Step](https://github.com/ghiscoding/angular-slickgrid/wiki/HOWTO---Step-by-Step).
- [X] Check that there isn't [already an issue](https://github.com/ghiscoding/Angular-Slickgrid/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/ghiscoding/Angular-Slickgrid/discussions).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by ghiscoding** [comment]
Please provide more details to reproduce, at the very least a print screen of the issue because at the moment I'm not exactly sure what you mean

**Comment by mcgolkev** [comment]
Firefox screenshot:

![Firefox Screenshot ](https://user-images.githubusercontent.com/24679283/200066869-e37ee73b-1a54-4175-8510-ace815875144.jpg)

Chromium screenshot:

![Chromium Screenshot](https://user-images.githubusercontent.com/24679283/200067022-bcac6939-dd94-4fb0-9659-faf37eacb173.jpg)

Unfortunately I could not figure out how to mouseover and activate the sort icon at the same time as taking a screen shot.  You may be able to notice in the Chromium screenshot that the gridmenu button is to the right of the column below it  whilst in the Firefox screen shot the button is directly over the column below it.  If I were to mouse over the menu button in Firefox, the sort icon would activate and become slightly visible around the edges of menu button.  In other words the menu button and sort icon occupy the same space on the page in Firefox.  In Chromium, the menu button is to the right in its own space and displays correctly.

Additionally the vertical scroll bar in Firefox is very narrow whereas the vertical scroll bar in Chromium is wider.  The menu icon is lined-up with the scroll bar in Chromium.  I'm not sure if the wider scroll bar in Chromium has anything to do with how the menu button is being displayed.

I hope this helps, I know it is not optimal, and I realize the version is very old.  I am a new dev and pasting in actual code to replicate this would be a bridge too far for me unfortunately. 

My hope was that there would be some grid options or some such thing that I could set in my code to have slickgrid render the menu button correctly in both browsers.  Thanks in advance for any help or direction.

**Comment by ghiscoding** [comment]
The only grid option related to this is `alwaysShowVerticalScroll` which is related to what you're described in issue #27 and was fixed in commit 5b45f3

**Comment by mcgolkev** [comment]
Thank you for your help!

I had already tried the alwaysShowVerticalScroll flag however it did not affect the problem.  We are not removing the vertical grid scroll, it just seemed coincidental that the scroll bars were rendered differently which is when I noticed the commit you mention above.  I am not able to provide a live repro.  I will certainly loop back if I find another way around this though.

Cheers.

**Comment by ghiscoding** [comment]
Showing the scroll was a hack that I found to avoid this issue, **however** if your styling theme makes the scroll thinner (for example the Material Design library does that), then the issue will be brought back. In other words, the scroll is a hack and for it to work its minimum width must match the grid menu icon, which is 18px I think

Like I said earlier, I was never able to find a better way to fix this issue, I would be happy to receive a fix in that regards

**Comment by mcgolkev** [comment]
It is possible that we are styling the scroll bar to be thinner.  That is a great place for me to research next.  Thanks again for all your help.  I'll let you know if I get a breakthrough!

**Comment by ghiscoding** [comment]
alright then, I'll close this issue as "won't fix" since at this moment there aren't any other ways to address this issue.

**Comment by ghiscoding** [comment]
@mcgolkev I forgot to mention, you could also use a smaller icon like the 3 vertical dots (as demoed in [Example 9](https://ghiscoding.github.io/Angular-Slickgrid/#/gridmenu)) which would probably work better with thinner scrollbar

![image](https://user-images.githubusercontent.com/643976/200105220-4f987cbe-9587-48dc-9b55-31b171444bb9.png)

**Comment by mcgolkev** [comment]
I have decided to create a button at a new location on the page and suppress the fa-bars in the left hand column.  I've been trying to follow Example 9 with no luck.  I have the following method linked to button event:

gridMenu(){
    const gridMenuInstance = this.angularGrid.extensionService.getExtensionByName(ExtensionName.gridMenu);
    gridMenuInstance.class.getAddonInstance;
  }
with no effect.

Additionally I have tried to use menuUsabilityOverride: (args) =>false in an attempt to prevent the gridMenu icon from appearing the left most column.  Again with no luck.

Is there a another example that  you might be able to share that goes into more detail about relocating the gridMenu button?

Thank you again for your help/direction/pity lol.

**Comment by ghiscoding** [comment]
The possible problem might be again related to the fact that you're using an old version of Angular-Slickgrid 3.x (I relaread 5.x a month ago) and I'm not sure if the Grid Menu code changed since that version. The best thing you can do is to check the Angular-Slickgrid-Demos project, most of the time I created a tag so that users still have ways to go to the old code, you can take a look at [Example 9 on v3.x](https://github.com/ghiscoding/angular-slickgrid-demos/blob/v3.3.0/bootstrap4-demo-with-translate/src/app/examples/grid-menu.component.ts) for the last 3.x version tag

Even though I provided a way to create an external Grid Menu button, we actually never went with that and kept the one in the grid and the one in the grid was created to always be on top right and cannot be moved to another location. The thing that I don't know and never tried is if we can disable/hide the one in the grid and only the external button, all the examples I created always had both showing at same time. If you want to hide the one in the grid, I assume you can just find it's CSS class name and just hide it with `display: none`

As a reminder, I stop supporting old versions since that is too much work for a single person (me) on a free basis (apart from the rare [Ko-fi](https://ko-fi.com/ghiscoding) that I receive). Also, there's just no way to keep copy of older version Wikis, so they might differ slightly, however using Angular-Slickgrid-Demos previous [tags](https://github.com/ghiscoding/angular-slickgrid-demos/releases) is your best way to look at real and see if anything changed

**Comment by mcgolkev** [comment]
That is great information and I very much appreciate you for taking the time re respond.  I'm going to advocate to upgrade to the new version and see where that gets me.

From a newbe's eyes this is like magic and very cool stuff.  I'm just soaking it all in like a sponge lol.

Thanks again for going the extra mile and being so responsive.

**Comment by ghiscoding** [comment]
@mcgolkev I took another look at this, because it's also a problem when using the grid option `autoHeight`. I couldn't find a good way to fix it directly in the lib because of all the styling that SlickGrid has, it seems hard to find a solution that works for everything. However I found that this code works well 

```scss
.slick-header-column:last-child {
    .slick-header-menu-button,
    .slick-resizable-handle {
        margin-right: 18px !important;
    }
}
```

Perhaps another approach, that I might look into, could be to hide the Grid Menu when hovering the column but show it back when hovering the Grid Menu, this actually might be better because column resizing would be aligned (while that is not the case with the patch above)

![brave_oFI9YJCXOo](https://user-images.githubusercontent.com/643976/206529888-c1a9dc31-86bb-4e61-89a5-f15807d3e5b6.gif)

**Comment by mcgolkev** [comment]
Thank you for following up Ghislain!  My Sr. Dev ended up solving this by pulling the sorting icon functionality from the grid and not dealing with it.  I'm planning to present this in a dev meeting  and see where that goes.

Thanks again your kind assistance.  Happy Holidays.

Cheers
